Assets/Scripts/Effects/WaterEffect.cs

```diff
@@ -0,0 +1,11 @@
+/**
+* Used for objects that react to water such as wind gust and plants. Generated by water material
+*/
+public class WaterEffect : AEffect {
+    public float WaterDensity {get; private set;}
+
+    public WaterEffect SetWaterDensity(float density) {
+        WaterDensity = density;
+        return this;
+    }
+}
```

Comment:
Can we use C#'s construction classes? Why have a private setter if you've exposed a setter method? Just do:
```suggestion
public class WaterEffect : AEffect {
    public float WaterDensity; // or { get; set; } 
}
```

Then construct with `new WaterEffect { WaterDensity = 5 }`.

Reply:
Same with all the other effect classes. I saw the DamageEffect class in an earlier PR but I think I passed it due to pressure. C# has a system for this and I think we should use it unless there's a good reason not to.

Reply:
Because we want this to be a builder pattern. The setters must return the object again

Reply:
Why